backend/s3_backend.go

```diff
@@ -196,6 +202,40 @@ func (s *S3Backend) Open(db, version, file string) (io.ReadCloser, error) {
 	return resp.Body, nil
 }
 
+func (s *S3Backend) Open(db, version, file string) (io.ReadCloser, error) {
+	// If this is a symlinked version, we should open the file at the
+	// target version directory.
+	targetVersion, err := s.getTargetVersion(db, version)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.open(db, targetVersion, file)
+}
+
+// getTargetVersion returns the target version of a symlinked version if the
+// _SYMLINK file exists or the same version otherwise.
+func (s *S3Backend) getTargetVersion(db, version string) (string, error) {
```

Comment:
I feel like since this gets called for every `Open`, is it worth it to cache results?

Reply:
Maybe have an option to skip this check? If we are just downloading files to load a version, no point in grabbing this file.

Reply:
I was thinking since the `Open` function is public and the caller has no knowledge of the fact that a version may be backed by a symlinked directory, we need to check with each call.